packages/nix/packages/pf-eks-resume.sh

```diff
@@ -0,0 +1,286 @@
+             \"template\": {
+               \"spec\": {
+                 \"schedulerName\": \"$scheduler_name\"
+               }
+             }
+           }
+  }"; then
+    echo "Successfully updated the scheduler to $scheduler_name for $deployment_name in namespace $namespace." >&2
+  else
+    echo "Failed to update the scheduler for $deployment_name." >&2
+  fi
+}
+
+# Function to scale up Karpenter node pools by removing limits
+scale_up_node_pools() {
+  # Check if the nodepool.karpenter.sh resource exists
+  if ! kubectl "$CONTEXT_ARGS" get nodepools.karpenter.sh -n "karpenter" &>/dev/null; then
+    echo "No nodepools.karpenter.sh resource found, skipping scaling down karpenter nodes." >&2
+    return
+  fi
+
+  for nodepool in $(kubectl "$CONTEXT_ARGS" get nodepools.karpenter.sh -n "karpenter" -o jsonpath='{.items[*].metadata.name}'); do
+    echo "Scaling up node pool: $nodepool" >&2
+    kubectl "$CONTEXT_ARGS" patch nodepool.karpenter.sh "$nodepool" -n "karpenter" --type='json' -p='[{"op": "remove", "path": "/spec/limits"}]'
+  done
+}
+
+clear_pending_pods() {
+  echo "Listing all Pending pods..." >&2
+
+  # Get all pods in 'Pending' state across all namespaces
+  pending_pods=$(kubectl "$CONTEXT_ARGS" get pods --all-namespaces --field-selector=status.phase=Pending -o json | jq -r '.items[] | "\(.metadata.namespace) \(.metadata.name)"')
+
+  if [ -z "$pending_pods" ]; then
+    echo "No Pending pods found." >&2
+  else
+    echo "Found Pending pods, starting deletion process..." >&2
+
+    # Loop through each pending pod and delete it
+    while read -r namespace pod_name; do
+      echo "Deleting pod: $pod_name in namespace: $namespace" >&2
+      kubectl "$CONTEXT_ARGS" delete pod "$pod_name" -n "$namespace"
+    done <<<"$pending_pods"
+
+    echo "All Pending pods have been deleted." >&2
+  fi
+}
+
+wait_for_panfactum_scheduler_pod() {
+  local namespace="scheduler"
+  local deployment_name="scheduler"
+
+  echo "Waiting for the scheduler pod to be scheduled and running in namespace $namespace..." >&2
+
+  # Timeout after 5 minutes (300 seconds)
+  timeout=300
+  interval=10
+  elapsed=0
+
+  while [[ $elapsed -lt $timeout ]]; do
+    pod_status=$(kubectl "$CONTEXT_ARGS" get pod -n "$namespace" -l panfactum.com/workload=scheduler -o jsonpath='{.items[0].status.phase}' 2>/dev/null)
+
+    if [[ $pod_status == "Running" ]]; then
+      echo "$deployment_name pod is running." >&2
+      return 0
+    else
+      echo "Pod status: $pod_status. Waiting..." >&2
+    fi
+
+    sleep $interval
+    elapsed=$((elapsed + interval))
+  done
+
+  echo "Timed out waiting for $deployment_name pod to be scheduled." >&2
+  return 1
+}
+
+restore_scheduler() {
+  local namespace="scheduler"
+  local deployment_name="scheduler"
+
+  deployment_exists=$(kubectl get deployments -n "$namespace" -o jsonpath="{.items[?(@.metadata.name=='$deployment_name')].metadata.name}")
+  if [ "$deployment_exists" == "$deployment_name" ]; then
```

Comment:
This will not work if we ever add another deployment to the scheduler namespace.

Reply:
@fullykubed wouldn't the deployment name be different in that case? What would be the failure?

Reply:
reviewed the query again I believe the jsonpath given unique deployment names in k8s ensures that this will be a single result.